pandas/core/indexes/period.py

```diff
@@ -514,6 +514,15 @@ def shift(self, periods: int = 1, freq=None) -> Self:
             )
         return self + periods
 
+    def _difference(self, other, sort=None) -> PeriodIndex:
+        if isinstance(other, Index) and other.inferred_type == "string":
+            try:
+                other = other.astype(self.dtype)
```

Comment:
I don’t think we do this type of casting in general. In fact, I’d expect the private _difference to only be reached with matching dtypes